libraries/ESP32/examples/Camera/CameraWebServer/app_httpd.cpp

```diff
@@ -1304,3 +1315,14 @@ void startCameraServer()
         httpd_register_uri_handler(stream_httpd, &stream_uri);
     }
 }
+
+void ledc_setup()
+{
+#ifdef CONFIG_LED_ILLUMINATOR_ENABLED
+  // PWM
+  ledcSetup(LEDC_CHANNEL_0, 4000, LEDC_TIMER_8_BIT);
+  ledcAttachPin(FLASH_LED_GPIO, LEDC_CHANNEL_0);
+#else
+  ESP_LOGI(TAG, "Flash led illumination is not configured");
+#endif
+}
```

Comment:
I think that all this code should go to `CameraWebServer.ino` instead of here.

Reply:
yes.

Reply:
> I think that all this code should go to `CameraWebServer.ino` instead of here.

Please take a look - CONFIG_LED_ILLUMINATOR_ENABLED is defined in a app_httpd.cpp file and MAY be undifined on compile time for CameraWebServer.ino since compile order may be various. So we have an issue in this case. This can be resolved by moving defines in the header file (as I suggest in the fist commit) but:
> Changing the example structure it outside of the scope of the issue.

So what is your desision for this case? Move all defines in the one CameraWebServer.ino or separate header or something else? 

cc: @me-no-dev

Reply:
Actually ... CONFIG_LED_ILLUMINATOR_ENABLED is defined nowhere.
I also searched it in sdkconfig and there is nothing there as well.

This is a sort of loose example, maybe coming from IDF. 

I changed my mind... just leave it there. I see other code lines in app_httpd.cpp that test CONFIG_LED_ILLUMINATOR_ENABLED.